scipy/spatial/transform/_rotation.pyx

```diff
@@ -1107,13 +1117,35 @@ cdef class Rotation:
         # If a single matrix is given, convert it to 3D 1 x 3 x 3 matrix but
         # set self._single to True so that we can return appropriate objects in
         # the `to_...` methods
-        cdef double[:, :, :] cmatrix
         if matrix.shape == (3, 3):
-            cmatrix = matrix[None, :, :]
+            matrix = matrix[np.newaxis, :, :]
             is_single = True
-        else:
-            cmatrix = matrix
 
+        # Calculate the determinant of the rotation matrix
+        # (should be positive for right-handed rotations)
+        dets = np.linalg.det(matrix)
+        if np.any(dets <= 0):
+            ind = np.where(dets <= 0)[0][0]
+            raise ValueError("Non-positive determinant (left-handed or null "
+                             f"coordinate frame) in rotation matrix {ind}: "
+                             f"{matrix[ind]}.")
+
+        # Gramian orthonormality check
+        # (should be the identity matrix for orthonormal matrices)
+        grams = matrix @ np.swapaxes(matrix, 1, 2)
+        orthonormal_ok = np.all(np.isclose(grams, np.eye(3), atol=1e-12), axis=(1, 2))
+        idxs = np.where(~orthonormal_ok | ~np.isclose(dets, 1, atol=1e-12))[0]
```

Comment:
What if we omit this block and will do orthogonalization regardless?

Reply:
I think it would be quite a bit slower for rotation matrices that are already well formed, as the expectation is that SVD is expensive relative to calculating these diagnostics. 

For the matrices where SVD is skipped, they will still be orthogonalized in the conversion to quaternion, and should be close enough that the original method is valid.

Reply:
Sounds good, I'll switch it over once this bugfix is in main.